File: 2540-minimum-common-value/2540-minimum-common-value.py
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        n1, n2= len(nums1), len(nums2)
        p1, p2=0, 0
        while p1<n1 and p2<n2:
            x=nums1[p1]
            y=nums2[p2]
            if x==y: return x
            elif x>y: p2+=1
            else: p1+=1
        return -1
        

File: 2540-minimum-common-value/2540-minimum-common-value.cpp
class Solution {
public:
    static int getCommon(vector<int>& nums1, vector<int>& nums2) {
        int n1=nums1.size(), n2=nums2.size();
        int p1=0, p2=0;
        for( ; p1<n1 && p2<n2; ){
            int x=nums1[p1], y=nums2[p2];
            if (x==y) return x;
            else if (x>y) p2++;
            else p1++;
        }
        return -1;
    }
};

auto init = []() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    return 'c';
}();